applications/compareSummary.cpp

```diff
@@ -0,0 +1,460 @@
+  int first_report2 = ecl_sum_get_first_report_step( ecl_sum2 );
+  int last_report2  = ecl_sum_get_last_report_step( ecl_sum2 );
+  
+  
+  //Calculates the time vectors
+  for(int report = first_report1; report <= last_report1; report++){
+
+    int time_index;
+    time_index = ecl_sum_iget_report_end( ecl_sum1 , report );
+    time_vec1.push_back( ecl_sum_iget_sim_days(ecl_sum1 , time_index));
+  }
+  
+  for(int report = first_report2; report <= last_report2; report++){
+    int time_index;
+    time_index = ecl_sum_iget_report_end( ecl_sum2, report );
+    time_vec2.push_back(ecl_sum_iget_sim_days(ecl_sum2 , time_index ));
+  }
+
+}
+
+void SummaryReader::setDataVecs(std::vector<double> &data_vec1,std::vector<double> &data_vec2, int index1, int index2){
+  //bool_vector_type * and int_vector_type * variables makes it possible to get the data corresponding to a certain keyword.
+  //
+  bool_vector_type  * has_var1   = bool_vector_alloc( stringlist_get_size( keys_short ), false );
+  int_vector_type   * var_index1 = int_vector_alloc( stringlist_get_size( keys_short ), -1 );
+  bool_vector_iset( has_var1 , index1 , true );
+  int_vector_iset( var_index1 , index1 , ecl_sum_get_general_var_params_index( ecl_sum_file_short , stringlist_iget( keys_short , index1) ));
+
+
+  bool_vector_type  * has_var2   = bool_vector_alloc( stringlist_get_size( keys_long ), false );
+  int_vector_type   * var_index2 = int_vector_alloc( stringlist_get_size( keys_long ), -1 );
+
+  bool_vector_iset( has_var2 , index2 , true );
+  int_vector_iset( var_index2 , index2 , ecl_sum_get_general_var_params_index( ecl_sum_file_long , stringlist_iget( keys_long , index2) ));
+
+  int first_report1 = ecl_sum_get_first_report_step( ecl_sum_file_short );
+  int last_report1  = ecl_sum_get_last_report_step( ecl_sum_file_short );
+  int first_report2 = ecl_sum_get_first_report_step( ecl_sum_file_long );
+  int last_report2  = ecl_sum_get_last_report_step( ecl_sum_file_long );
+  
+  //Calculates the data, function int_vector_iget() keeps track of which keyword to get data from.
+  for(int report = first_report1; report <= last_report1; report++){
+    int time_index;
+    time_index = ecl_sum_iget_report_end( ecl_sum_file_short , report );
+    data_vec1.push_back(ecl_sum_iget(ecl_sum_file_short, time_index, int_vector_iget( var_index1 , index1 )));
+  }
+
+  for(int report = first_report2; report <= last_report2; report++){
+    int time_index;
+    time_index = ecl_sum_iget_report_end( ecl_sum_file_long , report );
+    data_vec2.push_back(ecl_sum_iget(ecl_sum_file_long, time_index, int_vector_iget( var_index2 , index2 )));
+  }
+ 
+
+   
+}
+
+
+void SummaryReader::chooseReferance(std::vector<double> &time_vec1,std::vector<double> &time_vec2,std::vector<double> &data_vec1,std::vector<double> &data_vec2){
+  //This function is suppose to choose which vector that is referance vector, but when it is in use, get segmentation error
```

Comment:
I don't understand this comment.  What do you mean "when it is in use, get segmentation error"?  When the vector is in use or the function?